net/portmapper/portmapper.go

```diff
@@ -409,10 +414,18 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 		}
 	}
 
-	// And ask for a mapping.
-	pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
-	if _, err := uc.WriteTo(pmpReqMapping, pmpAddru); err != nil {
-		return netaddr.IPPort{}, err
+	if !haveRecentPMP && haveRecentPCP {
+		// Only do PCP mapping in the case when PMP did not appear to be available recently.
+		pkt := buildPCPRequestMappingPacket(myIP, localPort, prevPort, pcpMapLifetimeSec)
+		if _, err := uc.WriteTo(pkt, pcpAddru); err != nil {
+			return netaddr.IPPort{}, err
+		}
+	} else {
+		// And ask for a mapping.
```

Comment:
This comment style, staring with "And ask" no longer reads well in context of just the else.